mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/KeyRegistrationHandler.java

```diff
@@ -118,13 +118,13 @@ private void declareQueue(Sender sender) {
     }
 
     void stop() {
-        receiverSubscriber.filter(Predicate.not(Disposable::isDisposed))
-            .ifPresent(Disposable::dispose);
-        receiver.close();
         sender.delete(QueueSpecification.queue(registrationQueue.asString()))
             .timeout(TOPOLOGY_CHANGES_TIMEOUT)
             .retryWhen(Retry.backoff(retryBackoff.getMaxRetries(), retryBackoff.getFirstBackoff()).jitter(retryBackoff.getJitterFactor()).scheduler(Schedulers.elastic()))
             .block();
+        receiverSubscriber.filter(Predicate.not(Disposable::isDisposed))
+                .ifPresent(Disposable::dispose);
+        receiver.close();
```

Comment:
Nice catch!

Reply:
CI : "It's Over 9000!"

Reply:
Maybe this dude deserve a separated PR, we can open it in an extra PR to be merged faster don't you think?

Reply:
I was hoping that the build would pass but it just failed 2hours in on mailsShouldBeWellReceivedByBothRecipient{GuiceJamesServer} – org.apache.james.WithCassandraBlobStoreTest
(https://builds.apache.org/blue/organizations/jenkins/james%2FApacheJames/detail/PR-268/1/tests)
so yeah I will put a separate PR for the rabbitmq fix

Reply:
Just for the record, do you have an idea of which kind of INFRA this builds runs on? 

(cpu, memory, disk, virtualization setup?)

Reply:
I don't know about CPU and virtualization setup but the nodes seem to have 48GB of RAM and 4B of disk according to https://cwiki.apache.org/confluence/display/INFRA/Jenkins+node+labels

Reply:
> Maybe this dude deserve a separated PR, we can open it in an extra PR to be merged faster don't you think?

done in https://github.com/apache/james-project/pull/269